yaml/com/ibm/VPD/FRUManager.interface.yaml

```diff
@@ -0,0 +1,129 @@
+          keyword on hardware and also updates on the cache, if cache
+          information is present in the config JSON.
+      parameters:
+          - name: path
+            type: string
+            description: >
+                Inventory/hardware path of the FRU whose keyword needs to be
+                updated.
+          - name: data
+            type:
+                variant[struct[string,string,array[byte]],
+                struct[string,array[byte]]]
+            description: >
+                Contains data required to write. currently supports IPZ and
+                Keyword types VPD and more can be added.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+          - com.ibm.VPD.Error.EccCheckFailed
+
+    - name: ReadKeyword
+      description: >
+          Method to read VPD keyword value. Keyword can be read from hardware or
+          from cache.
+      parameters:
+          - name: path
+            type: string
+            description: >
+                Inventory/hardware Path of the FRU whose keyword needs to be
+                read.
+          - name: data
+            type: variant[struct[string,string],struct[string]]
+            description: >
+                Contains data required to read. currently supports IPZ and
+                Keyword type VPD and more can be added.
+          - name: target
+            type: boolean
+            description: >
+                Default will be false and will read from hardware in that case,
+                if true will read from cache.
+      returns:
+          - name: value
+            type: array[byte]
+            description: >
+                Value of the given keyword read from the given target.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+
+    - name: GetLocationCode
+      description: >
+          Method to get the expanded location code for the given inventory path.
+          If expanded location code is not available then unexpanded location
+          code will be returned.
+      parameters:
+          - name: inventoryPath
+            type: object_path
+            description: >
+                Inventory path of the FRU.
+      returns:
+          - name: locationCode
+            type: string
+            description: >
+                If expanded location code is available it returns expanded
+                location code else returns unexpanded location code.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+          - com.ibm.VPD.Error.PathNotFound
+
+    - name: deleteFRUVPD
+      description: >
+          Method to delete given FRU VPD and set its present property as false.
+      parameters:
+          - name: inventoryPath
+            type: object_path
+            description: >
+                Inventory path of the FRU.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+
+    - name: CollectFRUVPD
```

Comment:
The name does not match the description.
May be legacy, but please try to come up with something relevant now.